cyral/resource_cyral_integration_idp_saml.go

```diff
@@ -0,0 +1,165 @@
+package cyral
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/cyralinc/terraform-provider-cyral/client"
+)
+
+type CreateGenericSAMLRequest struct {
+	SAMLDraftId string                  `json:"samlDraftId"`
+	IdpMetadata *GenericSAMLIdpMetadata `json:"idpMetadata,omitempty"`
+
+	// Currently unused fields
+	IdpDescriptor *GenericSAMLIdpDescriptor `json:"idpDescriptor,omitempty"`
+}
+
+func (req *CreateGenericSAMLRequest) ReadFromSchema(d *schema.ResourceData) error {
+	req.SAMLDraftId = d.Get("saml_draft_id").(string)
+	if url := d.Get("idp_metadata_url").(string); url != "" {
+		req.IdpMetadata = &GenericSAMLIdpMetadata{
+			URL: url,
+		}
+	} else if xml := d.Get("idp_metadata_document").(string); xml != "" {
+		req.IdpMetadata = &GenericSAMLIdpMetadata{
+			XML: xml,
+		}
+	}
+	return nil
+}
+
+type CreateGenericSAMLResponse struct {
+	Integration GenericSAMLIntegration `json:"integration"`
+}
+
+func (resp *CreateGenericSAMLResponse) WriteToSchema(d *schema.ResourceData) error {
+	return resp.Integration.WriteToSchema(d)
+}
+
+func (resp *CreateGenericSAMLResponse) ReadFromSchema(d *schema.ResourceData) error {
+	return resp.Integration.ReadFromSchema(d)
+}
```

Comment:
Is this method used? I believe we can remove it.
```suggestion
```

Reply:
Done!